private/protocol/json/jsonutil/unmarshal.go

```diff
@@ -211,3 +227,33 @@ func unmarshalScalar(value reflect.Value, data interface{}, tag reflect.StructTa
 	}
 	return nil
 }
+
+func timeFromUnixString(d string) (time.Time, error) {
+	var secs, nsecs int64
+	if dot := strings.Index(d, "."); dot != -1 {
+		var err error
+		secs, err = strconv.ParseInt(d[:dot], 10, 64)
```

Comment:
The code you have there is not good, because float->int conversion rounds towards zero, so if the float approximation of the input millisecond is slightly less than the written value, this will subtract a whole ms, which isn't great.  https://play.golang.org/p/Ud_u9sFcS2

But in general, the idea of rounding to ms seems reasonable to me.  You could use [this function which is likely to be math.Round in Go 1.10](https://go-review.googlesource.com/c/43652/5/src/math/floor.go)

I don't suppose that (given that this library specifically targets AWS) there's something in the AWS protocol specs that specifies what precision is used for timestamps in each service?